File: Java/Remove Duplicates from Sorted Array.java
// 0ms, 100.00%; 41.5MB, 17.02%
class Solution {
    public int removeDuplicates(int[] nums) {
        int last = Integer.MIN_VALUE, j = 0;
        for (int i = 0; i < nums.length; ++i) {
            if (last != nums[i]) {
                nums[j++] = nums[i];
                last = nums[i];
            }
        }
        return j;
    }
}

File: Go/Remove Duplicates from Sorted Array.go
// 8ms, 90.55%; 4.6MB, 100.00%
func removeDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	length, last := 1, nums[0]
	for i := 1; i < len(nums); i++ {
		if nums[i] != last {
			nums[length] = nums[i]
			last = nums[i]
			length++
		}
	}
	return length
}
